.gitignore

```diff
@@ -0,0 +1,2 @@
+build
+*.pyc
```

Comment:
This is a good idea, but perhaps we should copy GitHub's gitignore for Python? https://github.com/github/gitignore/blob/main/Python.gitignore

I don't think we need every single line from that file, but I do think we could steal a few things. For example, [the first few lines](https://github.com/github/gitignore/blob/4488915eec0b3a45b5c63ead28f286819c0917de/Python.gitignore#L1-L4) cover several kinds of Python caches, which I think may get created when running `lit` here.

Reply:
Thanks for pointing me at that — I just added everything from https://github.com/github/gitignore/blob/main/Python.gitignore into our `.gitignore` and mentioned the origin reference.